src/kademlia/traversal_algorithm.cpp

```diff
@@ -607,10 +607,7 @@ void traversal_algorithm::look_for_nodes(bdecode_node const& r,bool const is_tra
 		while (end - nodes >= 20 + protocol_size + 2)
 		{
 			node_endpoint const nep = read_node_endpoint(protocol, nodes);
-			if(is_traverse)
-				traverse(nep.id, nep.ep);
-			else
-				get_node().m_table.heard_about(nep.id, nep.ep);
+			f(nep);
```

Comment:
I don't think you should give ``nep`` a name. just pass it straight into the function